scraper/circuit_breaker.py

```diff
@@ -0,0 +1,113 @@
+from collections import deque
+from diskcache import Deque
+import time
+import os
+
+def ShortCircuit(name, trigger=3, release=10, time_limit=120):
+    def decorator(fn):
+        breaker = CircuitBreaker(on=fn, name=name, trigger=trigger, release=release, time_limit=time_limit)
+        return breaker
+
+    return decorator
+
+# Circuit Breaker helper
+# When ON
+#  - delegates to the `on` parameter function
+#  - if `on()` fails it adds the failure to its count
+#  - if this count exceeds `trigger`, the breaker becomes OFF
+#  - if `on()` succeeds, it decrements the count
+# When OFF
+#  - delefates to the `off` parameter function
+#  - counts the numbers of times `off` is called
+#  - if this counts exceeds `release`, the breaker becomes ON
+class CircuitBreaker:
+    def __init__(self, name, on, off=None, trigger=3, release=10, time_limit=120):
+        self.policies = Deque(["ON" for _ in range(trigger)], f'/tmp/breaker/{name}')
```

Comment:
Ça devrait fonctionner pareil, non ?

```suggestion
        self.policies = Deque(["ON"] * trigger], f'/tmp/breaker/{name}')
```

Reply:
Haha probablement. Je suis pas 100% à l'aise avec ces raccourcis python. En vrai je suis plus pour extraire une fonction qui fait ça avait un nom. Pareil pour les += d'iterators sur un deque multiprocessor : trop de magie pour moi ^^